AsyncDisplayKit/Details/ASTraitCollection.h

```diff
@@ -0,0 +1,29 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <UIKit/UIKit.h>
+#import <AsyncDisplayKit/ASEnvironment.h>
+
+@interface ASTraitCollection : NSObject
+
+@property (nonatomic, assign) BOOL isMutable;
```

Comment:
Sounds good. FWIW, when ASViewController creates a ASTraitCollection via its override blocks, it immediately sets it to immutable (i.e., the user doesn't have to do it). This is similar to how layout specs get set to immutable. 

Though I'm not wild that isMutable is read/write so a user could just set it back to NO.